2019-10-26  Henry Kou  <henryk@lnxsrv06.seas.ucla.edu>

        * timespec.h: (timespec_cmp): Added a bug fix for integer overflow
        for the subtraction of a.tv_sec and b.tv_sec. Replaced with comparison
        operators between them and returning 1, -1, a.tv_nsec - b.tv_nsec if a.tv_sec
        is greater than b.tv_sec, if a.tv_sec is less than b.tv_sec, and else cases
        respectively.
--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec_fixed.h	2019-10-26 17:51:56.281903000 -0700
@@ -45,8 +45,14 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  //int diff = a.tv_sec - b.tv_sec;
+  if(a.tv_sec > b.tv_sec)
+    return 1;
+  else if(a.tv_sec < b.tv_sec)
+    return -1;
+  else
+    return a.tv_nsec - b.tv_nsec;
+  //return diff ? diff : a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
